File: Miners of the Broken Planet/MotBP JavaJason/Alien Miners Unity v2.5 (Demo) Versión Unithread/Assets/Scripts/Connection/Connection.cs
public abstract class Connection {

    private static Connection instance;

    public static Connection GetInstance() {
        if (instance == null)
        {
            instance = new ConnectionImp();
            Connection.GetInstance().Initialize();
        }
        return instance;
    }

    public abstract void Initialize();
    public abstract void Send(bool send, object ev);
    public abstract string Receive();
    public abstract void CloseConnection();
}

File: Alien Miners Uni-Jason/alien_miners_jason/src/java/connection/Connection.java
package connection;

public abstract class Connection{
	
	private static Connection instance;
		
    public static Connection getInstance() {
    	if (instance == null) {
    		instance = new ConnectionImp();
    	}
    	return instance; 
    }
		
	public abstract String receive() ;
	public abstract void send(String sentSentence);
	public abstract void stopConnection();
	public abstract ConnectionProperties getCp();    
}